sonar-batch/src/main/java/org/sonar/batch/bootstrap/ServerClient.java

```diff
@@ -90,80 +92,41 @@ public void download(String pathStartingWithSlash, File toFile, @Nullable Intege
     }
   }
 
-  public String request(String pathStartingWithSlash) {
-    return request(pathStartingWithSlash, GET, true);
-  }
-
-  public String request(String pathStartingWithSlash, String requestMethod) {
-    return request(pathStartingWithSlash, requestMethod, true);
+  public String downloadString(String pathStartingWithSlash) {
+    return downloadString(pathStartingWithSlash, GET, true, null);
   }
 
-  public String request(String pathStartingWithSlash, boolean wrapHttpException) {
-    return request(pathStartingWithSlash, GET, wrapHttpException, null);
-  }
-
-  public String request(String pathStartingWithSlash, String requestMethod, boolean wrapHttpException) {
-    return request(pathStartingWithSlash, requestMethod, wrapHttpException, null);
-  }
-
-  public String request(String pathStartingWithSlash, String requestMethod, boolean wrapHttpException, @Nullable Integer timeoutMillis) {
-    final byte[] buf = load(pathStartingWithSlash, requestMethod, wrapHttpException, timeoutMillis);
+  public String downloadString(String pathStartingWithSlash, String requestMethod, boolean wrapHttpException, @Nullable Integer timeoutMillis) {
+    InputStream is = load(pathStartingWithSlash, requestMethod, wrapHttpException, null, timeoutMillis);
     try {
-      return new String(buf, "UTF-8");
-    } catch (UnsupportedEncodingException e) {
+      return new String(IOUtils.toByteArray(is), "UTF-8");
+    } catch (IOException e) {
       throw new IllegalStateException(String.format("Unable to request: %s", pathStartingWithSlash), e);
     }
   }
 
-  public InputSupplier<InputStream> doRequest(String pathStartingWithSlash, String requestMethod, @Nullable Integer timeoutMillis) {
-    final byte[] buf = load(pathStartingWithSlash, requestMethod, false, timeoutMillis);
-
-    return new InputSupplier<InputStream>() {
-      @Override
-      public InputStream getInput() throws IOException {
-        return new ByteArrayInputStream(buf);
-      }
-    };
-  }
-
-  private byte[] load(String pathStartingWithSlash, String requestMethod, boolean wrapHttpException, @Nullable Integer timeoutMillis) {
-    Preconditions.checkArgument(pathStartingWithSlash.startsWith("/"), "Path must start with slash /");
-    String path = StringEscapeUtils.escapeHtml(pathStartingWithSlash);
-    URI uri = URI.create(getURL() + path);
+  /**
+   * @throws IllegalStateException on I/O error, not limited to the network connection and if HTTP response code > 400 and wrapHttpException is true
+   * @throws HttpException if HTTP response code > 400 and wrapHttpException is false
+   */
+  public InputStream load(String pathStartingWithSlash, String requestMethod, boolean wrapHttpException, @Nullable Integer connectTimeoutMs,
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Provide multiple methods instead of using "wrapHttpException" to determine which action to take. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS2301)